src/rust/engine/fs/src/store.rs

```diff
@@ -935,17 +926,20 @@ mod local {
         .inner
         .pool
         .spawn_fn(move || {
-          let (env, db, _) = dbs.get()?.get(&fingerprint);
+          let (env, db, _) = dbs.get()?.get(&digest.0);
           let ro_txn = env
             .begin_ro_txn()
             .map_err(|err| format!("Failed to begin read transaction: {}", err));
-          ro_txn.and_then(|txn| match txn.get(db, &fingerprint) {
-            Ok(bytes) => Ok(Some(f(Bytes::from(bytes)))),
+          ro_txn.and_then(|txn| match txn.get(db, &digest.0) {
+            Ok(bytes) => {
+              if bytes.len() == digest.1 {
+                Ok(Some(f(Bytes::from(bytes))))
+              } else {
```

Comment:
Does this case represent a hash collision...? If so, should we be noisier?

Reply:
Done